source/user-management.rst

```diff
@@ -0,0 +1,220 @@
+-------------------------------------------------
+
+In order to manually manage users, you can issue ``CREATE USER`` and ``GRANT`` statements as you normally would with any MySQL client connected to the cluster running in the context of the Operator. 
+
+We provide a special Docker image containing a client setup which is intended for running BASH and the MySQL client or executing SQL statements directly with ``kubectl run``. The name of this image is ``percona-client``.  
+
+An example of how you might use it to run a grant statement is as below:
+
+.. code:: bash
+ 
+    kubectl run -it --rm percona-client --image=percona:5.7 --restart=Never -- mysql -hcluster1-pxc -uroot -proot_password
+    mysql> GRANT ALL PRIVILEGES ON database1.* TO 'user1'@'%';
+
+You can refer to the official documentation to find out more about the `CREATE USER <https://dev.mysql.com/doc/refman/8.0/en/create-user.html>`_ and `GRANT <https://dev.mysql.com/doc/refman/8.0/en/grant.html>`_ statements.
+
+After you've created your users and issued grants, it's recommended to also manually sync these users to ProxySQL using ``proxysql-admin``.  The following one-liner can do this for you by running ``proxysql-admin --syncusers`` on the first ProxySQL instance in your cluster.
+
+.. code:: bash
+
+   kubectl exec -it cluster1-proxysql-0 -- proxysql-admin --config-file=/etc/proxysql-admin.cnf --syncusers
+
+You can validate a user synced successfully by connecting through ProxySQL with the MySQL client and then running a simple query.  A simple example of this is below:
+
+.. code:: bash
+
+   kubectl run -it --rm percona-client --image=percona:5.7 --restart=Never -- bash -il
+   percona-client:/$ mysql -h cluster1-proxysql -uuser1 -ppassword1
+   mysql> SELECT * FROM database1.table1 LIMIT 1;
+
+Method 2: Automatically Manage Users with Kubernetes Secret
+-----------------------------------------------------------
+
+YAML Object Format
+******************
+
+The YAML Object is stored as a `Kubernetes Secrets <https://kubernetes.io/docs/concepts/configuration/secret/>`_ object, containing two sets of data:
+
+1. Some **username:password** pairs that are compatible with `Kubernetes Selectors <https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/>`_,
+2. A `stringData <https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret-manually>`_ field with a specially composed and formatted text.
+
+The text stored in ``stringData`` field is a *YAML fragment*, which contains two *dictionaries of dictionaries*: 
+
+1. **roles dictionary** specifies role grants,
+2. **users dictionary** specifies user grants.
+
+The following example illustrates the whole YAML object:
+
+.. code:: yaml
+
+   apiVersion: v1
+   kind: Secret
+   metadata:
+     name: secrets-for-users
+     annotations:
+       status: applied
+       checksum:
+   type: Opaque
+   stringData:
+     myuser1: password1
+     myuser2: password2
+     grants.yaml: |-
+       roles:
```

Comment:
can we put `roles` under `users` section?
reason - increase readability of the example